File: Algorithms/Excel Sheet Column Number.js
// https://leetcode.com/problems/excel-sheet-column-number/

/**
 * @param {string} s
 * @return {number}
 *
 * Given s='AB', return 28
 */
var titleToNumber = function(s) {
  var i = s.length,
      result = 0,
      base = 'A'.charCodeAt(0) - 1,
      base1 = s.length - 1
  while (--i >= 0) {
    result += (s.charCodeAt(i) - base) * Math.pow(26, (base1-i))
  }
  return result
};

console.log(titleToNumber('ABCD'))

File: Go/171-excel-sheet-column-number/excel_sheet_column_number.go
package title

import (
	"math"
)

func titleToNumber(s string) int {
	ret := 0
	for i := len(s) - 1; i >= 0; i-- {
		base := math.Pow(26, float64(len(s)-i-1))
		ret += int(s[i]-'A'+1) * int(base)
	}
	return ret
}
